server/src/main/java/org/apache/druid/server/QueryResultPusher.java

```diff
@@ -120,7 +125,9 @@ public Response push()
 
       final Response.ResponseBuilder startResponse = resultsWriter.start();
       if (startResponse != null) {
-        startResponse.header(QueryResource.QUERY_ID_RESPONSE_HEADER, queryId);
+        startResponse.header(QueryResource.QUERY_ID_RESPONSE_HEADER, queryId)
```

Comment:
can you point me to the RFC that mandates populating this field? From what I read, its only required when you want the receiver to prep for that trailer field .e.g checksum. In this case, we could send just the "response complete" trailer header so that clients will know that they can expect an error message at the end if the response is truncated.

Reply:
This may be https://datatracker.ietf.org/doc/html/rfc7230#section-4.4

Reply:
It doesn't say that you must specify this header. Only when you want the caller to know beforehand that such a trailer will be added. We can remove this header entirely.

Reply:
Right, it is not mandatory. So if it not needed we can remove it. Just double checking if `caller to know beforehand` is not something we can skip. Let me know

Reply:
You could just add the `RESPONSE_COMPLETE_TRAILER_HEADER`. Callers can check if that header is present and use it to detect truncated response.

Reply:
So you meant we should not add this header at all when response is trunked?

Reply:
Ah no. This header would not mean that the response is truncated. It would mean that if the response is truncated, they can expect a corresponding trailer. so client logic can be something like below
start reading response 
if response_complete _trailer header is present 
  use the trailer logic to detect truncated response
else
  use the existing mechanism to detect truncated response

Reply:
Done